sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -516,6 +516,7 @@ case class NewInstance(
 
   override def nullable: Boolean = needNullCheck
 
+  override def foldable: Boolean = false
```

Comment:
Good catch! I think this change makes sense. `NewInstance` should create a fresh instance per invocation. Can we add some code comments here?

Reply:
I will add a comment.